app/models/user_role.rb

```diff
@@ -11,6 +11,18 @@ class UserRole < ActiveRecord::Base
   FACILITY_STAFF = "Facility Staff".freeze
   FACILITY_SENIOR_STAFF = "Facility Senior Staff".freeze
 
+  module AssociationExtension
+
+    def operator?(facility)
+      any? { |ur| ur.facility == facility && ur.in?(facility_roles) }
+    end
+
+    def manager?(facility)
+      any? { |ur| ur.facility == facility && ur.in?(facility_management_roles) }
+    end
+
```

Comment:
I'm not 100% what rails is doing under the hood, but I believe every time you call `user.operator_of?`, it's treating `user.user_roles.any?` as a new query, so it doesn't memoize `user.user_roles`. Moving this into an extension seems to handle the memoization. http://guides.rubyonrails.org/association_basics.html#association-extensions